src/stream.jl

```diff
@@ -1,3 +1,29 @@
+const RETRYABLE_CODES = (
+    "IncompleteSignature",
+    "ThrottlingException",
+    "RequestExpired",
+)
+
+function aws_retry_cond(s, e)
+    if e isa AWSException && (500 <= e.cause.status <= 504 || e.code in RETRYABLE_CODES)
+        debug(LOGGER, "CloudWatchLogs operation encountered $(e.code); retrying")
+        return (s, true)
+    elseif e isa MbedException
+        debug(LOGGER, "CloudWatchLogs operation encountered $e; retrying")
+        return (s, true)
+    end
+
+    return (s, false)
+end
+
+function aws_retry(f)
+    wrapped = retry(delays=GENERIC_AWS_DELAYS, check=aws_retry_cond) do
```

Comment:
Just FYI you're effectively doing `retry(()->f(), ...)`, which can be written as just `retry(f, ...)`.